pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1064,7 +1084,27 @@ func (vs *VSphere) DetachDisk(volPath string, nodeName string) error {
 }
 
 // CreateVolume creates a volume of given size (in KiB).
-func (vs *VSphere) CreateVolume(name string, size int, tags *map[string]string) (volumePath string, err error) {
+func (vs *VSphere) CreateVolume(volumeOptions *VolumeOptions) (volumePath string, err error) {
+
+	// Default diskformat as 'thin'
+	if volumeOptions.DiskFormat == "" {
+		volumeOptions.DiskFormat = ThinDiskType
+	}
+
+	if diskFormatValidType[volumeOptions.DiskFormat] == "" {
```

Comment:
this could consolidate to 
```go
if diskformat, ok := diskFormatValidType[volumeOptions.DiskFormat]; !ok {
 return "", fmt.Errorf(".....")
}
```
also pre-create the valid disk format options string and error to save the loop